datalad/config.py

```diff
@@ -275,6 +281,19 @@ def reload(self, force=False):
         # override with environment variables
         self._store = _parse_env(self._store)
 
+        if not ConfigManager._checked_git_identity:
+            for cfg, envs in (
+                    ('user.name', ('GIT_AUTHOR_NAME', 'GIT_COMMITTER_NAME')),
+                    ('user.email', ('GIT_AUTHOR_EMAIL', 'GIT_COMMITTER_EMAIL'))):
+                if cfg not in self._store \
+                        and not any(e in os.environ for e in envs):
+                    lgr.warning(
+                        "It is highly recommended to configure Git before using "
+                        "DataLad. Set both 'user.name' and 'user.email' "
+                        "configuration variables."
+                    )
+            ConfigManager._checked_git_identity = True
```

Comment:
`s/ConfigManager/self/` feels a bit more natural to me, but it's of course fine as is.

Reply:
True, but I my personal preference is pointing the other way ;-)